airflow/providers/google/ads/transfers/ads_to_gcs.py

```diff
@@ -92,6 +94,7 @@ def __init__(
         page_size: int = 10000,
         gzip: bool = False,
         impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        api_version: str = "v5",
```

Comment:
what might be better is to make this `Optional[str] = None` (for all the operators)

then only supply to hook if provided

this way, when we need to increment the version again, we only need to do it in one place.

Reply:
I agree with @dstandish. Using optional and hook's default value as fallback would be best - we will have only one place to change in future.

Reply:
Somehow missed the notification. I agree, this way is more elegant.

Reply:
Since passing in `None` is not the same as not passing in any parameter, I will have to change `GoogleAdsHook` to something like this. Does this look ok?
```python
    def __init__(
        self,
        api_version: str,
        gcp_conn_id: str = "google_cloud_default",
        google_ads_conn_id: str = "google_ads_default",
    ) -> None:
        super().__init__()
        self.api_version = api_version if api_version else "v5"
        ...
```
This doesn't feel Pythonic, but it does allow us to only change one place in the future.

I wanted to see how other provider hooks deal with `api_version`
1. [ComputeEngineHook](https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/hooks/compute.py#L52) takes `api_version` as an argument, but [ComputeEngineSSHHook](https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/hooks/compute_ssh.py#L144) which calls ComputeEngineHook doesn't pass in `api_version`.
2.  [CloudSQLHook](https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/hooks/cloud_sql.py#L96) takes `api_version` but has no default value. Instead [CloudSQLCreateInstanceOperator](https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/operators/cloud_sql.py#L320) has a default value assigned.

It seems like there is no set method for this specific issue.

Reply:
i see, thinking about how to handle case if operator calls hook with version None

but i would make it so the operator doesn't pass that param to the hook unless it's not None.  then in that case it uses hook default.

then in the hook you can do it in the normal way
```python
    def __init__(
        self,
        api_version: str='v5',
        gcp_conn_id: str = "google_cloud_default",
        google_ads_conn_id: str = "google_ads_default",
    ) -> None:
        super().__init__()
        self.api_version = api_version
        ...
```

if for some reason it is not possible to do this with the operators (unlikely) i would do this:

```python
    def __init__(
        self,
        api_version: Optional[str]=None,
        gcp_conn_id: str = "google_cloud_default",
        google_ads_conn_id: str = "google_ads_default",
    ) -> None:
        super().__init__()
        self.api_version = api_version or 'v5'
        ...
```

true it's a bit hidden, and that's why first suggestion is better, but it's still better than setting it in many places

if you wanted to make it more brightly visible, you could consider defining a constant in the module `DEFAULT_API_VERSION` and replace `'v5'` with that

Lastly, with some clients, not supplying api version means will use latest. and if that were to be true here, i'd suggest doing that, so we don't need any version identifier in the code.  But i think it may not work that way with gcp.

Reply:
Sorry for keep asking questions about a seemingly simple issue. However, when you say
> I would make it so the operator doesn't pass that param to the hook unless it's not None.

do you mean something like this?
```python
        if self.api_version:
            ads_hook = GoogleAdsHook(
                gcp_conn_id=self.gcp_conn_id,
                google_ads_conn_id=self.google_ads_conn_id,
                api_version=self.api_version,
            )
        else:
            ads_hook = GoogleAdsHook(
                gcp_conn_id=self.gcp_conn_id,
                google_ads_conn_id=self.google_ads_conn_id,
            )
```

Reply:
no worries at all

so i think the standard appaoach is something like this:
```python
            kwargs = {}
            if self.api_version:
                kwargs.update(api_version=self.api_version)
            ads_hook = GoogleAdsHook(
                gcp_conn_id=self.gcp_conn_id,
                google_ads_conn_id=self.google_ads_conn_id,
                **kwargs,
            )
```